universal_doc_importer/file_system.py

```diff
@@ -37,10 +48,64 @@ def printAllChildren(self, depth=-1):
                 child.printAllChildren(depth)
             print("\t" * depth + "}")
 
-    def makeDict(self):
-        dictionary = {self.name: []}
+    def buildDict(self):
+        """
+        Recursive function that Built dict from file names of the repo:
+        :param: None:
+        :return: data:
+        :rtype: dict:
+        """
         if len(self.children) > 0:
+            dictionary = {self.name: []}
             for child in self.children:
-                dictionary[self.name].append(child.makeDict())
+                dictionary[self.name].append(child.buildDict())
+            return dictionary
+        else:
+            return self.name
+
+    def makeDict(self):
+        """
+        Make filtered data
+        :param: None:
+        :return: data:
+        :rtype: dict:
+        """
+        #  Get data from buildDict method and check if data is empty
+        #  it returns data like this { repo_name : []}
+        data = self.buildDict()
+        repo_name = '/'.join(list(data)[0].split('-')[:2])
+        result = self.filter_by_extension(data)
+        if result == {}:
+            result[repo_name] = []
```

Comment:
Once the `_buildDict` is returning dict anyway, you don't need this. self.name should be filled with the repo name somewhere in the `__init__`